src/spamfilter.c

```diff
@@ -488,16 +488,33 @@ void stripcolors(char new[512], char *org)
 {
     int len = 0;
 
-    for(; (*org && len<512); org++)
+    for(; (*org && len < 512); org++)
     {
-        if(*org=='\003')
+        if(*org == '\003')
         {
+            // Color codes: 1-2 digits, then optionally followed by a comma and an additional 1-2 digits
             org++;
-            while(IsDigit(*org) || *org==',')
+            if(*org && IsDigit(*org))
+            {
                 org++;
+                if(*org && IsDigit(*org))
+                {
```

Comment:
From a readability perspective, all these extra brackets are... extra.  If your scoped statement is only one line long, shouldn't be using brackets...